api/src/application/user/user-routes.js

```diff
@@ -214,33 +214,7 @@ module.exports = (service, controller, mixedValidation, ValidationError, server,
           }, {
             client_id: clientValidator,
           }),
-          payload: Joi.object().keys({
-            name: Joi.string().allow(''),
-            given_name: Joi.string().allow(''),
-            family_name: Joi.string().allow(''),
-            middle_name: Joi.string().allow(''),
-            nickname: Joi.string().allow(''),
-            preferred_username: Joi.string().allow(''),
-            profile: Joi.string().uri().allow(''),
-            shouldClearPicture: Joi.boolean(),
-            picture: Joi.object().type(Readable).assert(
-              'hapi.headers.content-type',
-              Joi.any().valid(['image/jpeg', 'image/png', 'application/octet-stream'])
-            ),
-            website: Joi.string().uri().allow(''),
-            email: Joi.string().email().allow(''),
-            gender: Joi.string().allow(''),
-            birthdate: Joi.string().isoDate().allow(''),
-            zoneinfo: Joi.string().valid(userFormData.timezones),
-            locale: Joi.string().valid(Object.keys(userFormData.locales)),
-            phone_number: Joi.string().allow(''),
-            'address.street_address': Joi.string().allow(''),
-            'address.locality': Joi.string().allow(''),
-            'address.region': Joi.string().allow(''),
-            'address.postal_code': Joi.string().allow(''),
-            'address.country': Joi.string().allow(''),
-          }).invalid(null).label('picture') // null payload means picture size validation failed
```

Comment:
This is the only thing that makes me worry about this approach. Did you mess around with a big ol' image and see how this endpoint behaves with a giant image? I bet it deletes your whole profile.

I think there's a way to configure validation to allow keys that aren't explicitly set in hapi... it'll take a little documentation digging to find though I'll link it if/when I find it.

The other alternative is to watch for a null payload and manually return a boom error

Reply:
Here's an issue talking about it https://github.com/hapijs/hapi/issues/1308

You should be able to do that just for this route